### PR TITLE
spark-3.0.1 compatible

### PR DESCRIPTION
**Pull request by vanhoale**

### CODE REVIEW

**Review by jsleight — COMMENTED** [review]
how does this compare to #70? (just making sure I understand the differences 😄 )

I think this is a subset of that work (except the spark v3.0.0 vs 3.0.1).  This is a much smaller PR, so is ok to be a subset.

File: version.sbt
Comment (jsleight) [inline version.sbt:1]: @lucagiovagnoli what are thoughts w.r.t. just releasing this as a minor/patch upgrade rather than adding a suffix?

**Comment by vanhoale** [comment]
> how does this compare to #70? (just making sure I understand the differences 😄 )
> 
> I think this is a subset of that work (except the spark v3.0.0 vs 3.0.1). This is a much smaller PR, so is ok to be a subset.

Hi @jsleight thanks for looking at this PR and for pointing me to the #70 . My PR is just a simple spark version upgrade, it may help to resolve the #76 quickly, otherwise, we can wait for #70 release

File: build.sbt
Comment (vanhoale) [reply build.sbt:47]: thanks, I will remove it

Comment (lucagiovagnoli) [reply build.sbt]: Why removing it? Can there be people using scala 2.12 and spark 3 ?

File: .travis.yml
Comment (lucagiovagnoli) [inline .travis.yml:16]: should we keep testing for this ?

File: build.sbt
Comment (vanhoale) [reply build.sbt]: @lucagiovagnoli this is a good point since spark 3 will no longer support scala 2.11. So maybe we should have a new release branch for spark 3 compatible?

File: version.sbt
Comment (vanhoale) [reply version.sbt:1]: the version may be related to the comment above

File: build.sbt
Comment (jsleight) [reply build.sbt]: Yeah, can remove cross publish because scala 2.11 isn't compatible with spark 3.  Will only be publishing scala 2.12 versions.

**Review by lucagiovagnoli — COMMENTED** [review]
Looks good! What kind of testing have you performed ?

File: .travis.yml
Comment (vanhoale) [reply .travis.yml:16]: Yes, this is also my concern, spark 2.x needs this to be passed. What do you think?

**Comment by vanhoale** [comment]
hi @lucagiovagnoli so we can't test the same code base with 2 spark api versions

File: build.sbt
Comment (lucagiovagnoli) [reply build.sbt]: Oh right yes. So I'd be in favor of a separate release branch for spark 3, but I'd be worried about maintenance overhead. Maybe we can have master "follow" spark 3 and still accept PRs on a branch pointing to the latest commit using spark 2.x

File: .travis.yml
Comment (lucagiovagnoli) [reply .travis.yml:16]: Let's move on entirely on spark 3 in this PR.

File: .travis.yml
Comment (lucagiovagnoli) [inline .travis.yml:13]: remove space

**Comment by lucagiovagnoli** [comment]
> Looks good! What kind of testing have you performed ?

Gentle bump on this question. It seems to me that moving to spark3 required an awful small amount of code and my gut feeling tells me that something is wrong. Do all tests pass ? I will also try running this against our internal integration suite

**Review by vanhoale — COMMENTED** [review]
> > Looks good! What kind of testing have you performed ?
> 
> Gentle bump on this question. It seems to me that moving to spark3 required an awful small amount of code and my gut feeling tells me that something is wrong. Do all tests pass ? I will also try running this against our internal integration suite

yes, it seems all the integration suites completed successfully when I did run `sbt it:test` locally. Does travis CI do the same?

**Comment by vanhoale** [comment]
> > > Looks good! What kind of testing have you performed ?
> > 
> > 
> > Gentle bump on this question. It seems to me that moving to spark3 required an awful small amount of code and my gut feeling tells me that something is wrong. Do all tests pass ? I will also try running this against our internal integration suite
> 
> yes, it seems all the integration suites completed successfully when I did run `sbt it:test` locally. Does travis CI do the same?

```
[info] IAMIntegrationSuite:
[info] - roundtrip save and load !!! IGNORED !!!
[info] - load fails if IAM role cannot be assumed !!! IGNORED !!!
[info] PostgresDriverIntegrationSuite:
[info] - postgresql driver takes precedence for jdbc:postgresql:// URIs !!! IGNORED !!!
[info] - roundtrip save and load !!! IGNORED !!!
20/10/07 18:05:57 INFO ShutdownHookManager: Shutdown hook called
20/10/07 18:05:57 INFO ShutdownHookManager: Deleting directory /private/var/folders/h9/w3kwqgt52vd6044pmp_kglx817hfbk/T/spark-db2c39fe-aa22-43e7-8f30-5b65fa1ba91e
[info] Run completed in 44 minutes, 44 seconds.
[info] Total number of tests run: 48
[info] Suites: completed 11, aborted 0
[info] Tests: succeeded 48, failed 0, canceled 0, ignored 7, pending 0
[info] All tests passed.
[success] Total time: 2688 s, completed Oct 7, 2020 6:05:57 PM
```

**Comment by lucagiovagnoli** [comment]
Travis CI doesn't run it:test iirc but I've also just manually verified internally that `it:test` works. So this PR looks good to me. Before merging it in and publishing to nexus, @jsleight, @lydian  how do you feel about `4.2.1` vs `5.0.0` ?

**Comment by jsleight** [comment]
> Travis CI doesn't run it:test iirc but I've also just manually verified internally that `it:test` works. So this PR looks good to me. Before merging it in and publishing to nexus, @jsleight, @lydian how do you feel about `4.2.1` vs `5.0.0` ?

Minor version update imo, so `4.2.0`.  Our public API is unchanged, and this [faq on semver page](https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api) suggests that updates to package dependencies should be minor or patch in these cases.  I view this as a rather significant dependency update, so minor version bump seems correct?

File: version.sbt
Comment (jsleight) [inline version.sbt:1]: 4.2.0 would be the version 😄

**Review by jsleight — APPROVED** [review]

File: version.sbt
Comment (lucagiovagnoli) [reply version.sbt:1]: That's right. I'll wait on this last small update and merge this in right after

**Review by lucagiovagnoli — APPROVED** [review]

**Comment by lucagiovagnoli** [comment]
This is released on sonatype

**Comment by vanhoale** [comment]
> This is released on sonatype

Thanks @lucagiovagnoli

**Pull request merged**
